script.js

```diff
@@ -1 +1,252 @@
+  // createCard(cardElement);
+  // board.column.row.appendChild(createCard);
+  // board[column][row].appendChild(createCard);
+
+  const clickedCard = board[column][row];
+
+  console.log('FIRST CARD', firstCard);
+  console.log('CLICKED CARD', board[column][row]);
+
+  // if no first card has been selected,
+  if (firstCard === null) {
+    // assign clicked card data to firstCard variable
+    firstCard = clickedCard;
+    // assign the clicked card's element to the firstCardElement variable
+    firstCardElement = cardElement;
+    // turn this card over; set card element's inner text to display the card's name
+    cardElement.innerText = firstCard.display;
+    outputBox.innerText = 'Choose a second card';
+    // if user attempts to re-click the same square, tell him to click other squares
+  } else if (firstCardElement == cardElement) {
+    outputBox.innerText = 'Pls pick a different card';
+  }
+
+  // if the cards match: if the property 'name' of the clicked card is  the same as the first card's
+  else if ((clickedCard.display === firstCard.display
+  // if the property 'suit' of the clicked card is the same as the first card's
+  && clickedCard.suit === firstCard.suit)) {
+    // turn this card over
+    console.log('match');
+    // change inner text of cardElement variable to show the 'name' property in firstCard
+    cardElement.innerText = firstCard.display;
+    outputBox.innerText = 'Match! You win!';
+    setTimeout(() => {
+      outputBox.innerText = '';
+    }, 3000);
+    firstCard = null;
+  }
+  // if the cards do not match:
+  else {
+    canClick = false;
+    cardElement.innerText = clickedCard.display;
+    setTimeout(() => {
+    // clear the data of the first card
+      firstCard = null;
+      // make the card data disappear from user's view
+      firstCardElement.innerText = '';
+      cardElement.innerText = '';
+
+      // re-enable clicking
+      canClick = true;
+    }, 2000);
+    outputBox.innerText = 'No match! :( Try again';
+  }
+};
+
+// ------------build the board's elements-------------------------------
+const buildBoardElements = (board) => {
+  // create the element that everything will go inside of
+  const boardElement = document.createElement('div');
+
+  // give it a class for CSS purposes
+  boardElement.classList.add('board');
+
+  // use the board data structure we passed in to create the correct size board
+  for (let i = 0; i < board.length; i += 1) {
+    // make a var for just this row of cards
+    const row = board[i];
+
+    // make an element for this row of cards
+    const rowElement = document.createElement('div');
+    rowElement.classList.add('row');
+
+    // make all the squares for this row
+    for (let j = 0; j < row.length; j += 1) {
+      // create the square element
+      const square = document.createElement('div');
+
+      // set a class for CSS purposes
+      square.classList.add('square');
+
+      // set the click event
+      // eslint-disable-next-line
```

Comment:
I think disabling eslint on this line was a mistake- i see the formatting is weird. Why are we disabling eslint on this line? Could we add a comment every time we disable eslint to explain why?